cli/command/image/build/context.go

```diff
@@ -89,7 +89,7 @@ func ValidateContextDirectory(srcPath string, excludes []string) error {
 func DetectArchiveReader(input io.ReadCloser) (rc io.ReadCloser, isArchive bool, err error) {
 	buf := bufio.NewReader(input)
 
-	magic, err := buf.Peek(archiveHeaderSize)
+	magic, err := buf.Peek(archiveHeaderSize * 2)
```

Comment:
i'm surprised this works, because magic should be twice as big, which ends up going to `archive.DetectCompression` which looks for byte equality in the beginning of the magic https://github.com/docker/cli/blob/master/vendor/github.com/docker/docker/pkg/archive/archive.go#L147 Am I missing something ?

Reply:
Or I guess it would just skip it, because of the different length https://github.com/docker/cli/blob/master/vendor/github.com/docker/docker/pkg/archive/archive.go#L143

Reply:
Nevermind I thought this was for the compressed codepath, it's not.